### PR TITLE
Check if instances are up before creating new ones.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1001/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1002/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -3827,37 +3827,45 @@ def spawn_and_setup_appengine()
...
-    options = @options.to_a.flatten
     initialize_nodes_in_parallel(appengine_info)
   end
 
-  def spawn_appengine(nodes)
-    Djinn.log_debug("nodes is #{nodes.join(', ')}")
-    return [] if nodes.length.zero?
-
+  def spawn_appengine(machines)
+    Djinn.log_debug("nodes is #{machines.join(', ')}")
```
Comment (cdonati) [inline AppController/djinn.rb]: This log statement might be confusing now that you renamed the variable.

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1010/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1012/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1014/

File: AppController/djinn.rb [hunk 2]
```diff
@@ -3805,48 +3805,58 @@ def spawn_and_setup_appengine()
...
+      # if the instances have been already spawned: we can do that
+      # comparing what we are requested and what we have in @nodes.
+      # Note: the tools doesn't include the headnode in machines.
+      if @nodes.length < (machines.length + 1)
+        @state = "Spawning up #{machines.length} virtual machines"
+        roles = machines.map { |node| node['jobs'] }
+        disks = machines.map { |node| node['disk'] }
+
+        Djinn.log_info("Starting #{machines.length} machines.")
+        # since there's only one cloud, call it cloud1 to tell us
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Full sentences please.

File: AppController/djinn.rb [hunk 3]
```diff
@@ -3805,48 +3805,59 @@ def spawn_and_setup_appengine()
...
     else
-      nodes.each { |node|
+      # For cluster mode, we return a nodes structure with the correct
+      # values, since we already have the jobs and ips for the layout.
+      machines.each { |node|
         appengine_info << {
           'public_ip' => node['ip'],
           'private_ip' => node['ip'],
           'jobs' => node['jobs'],
-          'instance_id' => 'i-SGOOBARZ',
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Do we have unit tests that break because of this change?

Comment (obino) [inline AppController/djinn.rb]: OCD was clean, so no unit test were harm in the making of this patch :)

**Pull request merged**